crates/core/src/kernel/arrow/engine_ext.rs

```diff
@@ -504,6 +504,28 @@ impl<T: ExpressionEvaluator + ?Sized> ExpressionEvaluatorExt for T {
     }
 }
 
+pub trait StructDataExt {
```

Comment:
You're killin me here with the lack of rustdocs :grimacing: 

I understand what this trait is but I don't understand why it must exist and when it can go away

 I'm going to add a CI check :smiling_imp:

Reply:
I provosioned a little help and added more RustDocs ... I'll keep working on myself 😆